src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java

```diff
@@ -414,10 +422,18 @@ private void handleClientRoles(RoutingContext rc) {
             // upload to storage
             storeWriter.upload(clients, null);
 
+            ensurePublisherSiteKeyExists(c);
```

Comment:
I wonder if there's any deduplication we could do here. eg add & update  both upload/ensure/respond, and future changes to these would happen in sync